snagsby/sources.py

```diff
@@ -96,9 +136,13 @@ def _parse_sources_str(sources_str):
         for source in SPLITTER.split(sources_str) if source
     ]
 
+def parse_source(source):
+    if not re.match(r's[m3]:\/\/', source):
+        logger.debug('Sources must start with s3:// or sm://')
+    elif source.startswith('s3://'):
+        return S3Source(source)
+    else:
+        return SMSource(source)
 
 def parse_sources(sources_str):
-    return [
-        S3Source(source)
-        for source in _parse_sources_str(sources_str)
-    ]
+    return list(filter(None, map(parse_source, _parse_sources_str(sources_str))))
```

Comment:
We tend to favor comprehensions over map unless a comprehension is totally unreasonable. Either way, we shouldn't need the outer `list()` call. Maybe?:

```
    parsed_sources = [
        parse_source(source)
        for source in _parse_sources_str(sources_str)
    ]
    return filter(None, parsed_sources)
```

Reply:
list is still needed but I made it a comprehension